### DISCUSSION TITLE
Persistent Data in store and forward

### DISCUSSION THREAD

**Discussion by cityoke2** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#78] [Q&A]
Thank you very much for your work. I wanted ask what steps need to be taken so that historical Data is not lost when node red is restarted before it has reconnected with the host application. I have enabled node red persistence to the local storage module but I seem to be still losing data on restart if the device module was buffering data.

**Comment by thomassorensen2650** [comment]
Have you tried to disconnect the host application and see if a context file is created? It only persist to disk every 30 second by default, so you can loose up to 30 seconds of data with the default configuration.

https://nodered.org/docs/api/context/store/localfilesystem#options
